### PR TITLE
Add eureka service discovery

### PR DESCRIPTION
**Pull request by andrsuh**

### CODE REVIEW

**Review by AfanasievYuriy — DISMISSED** [review]

**Comment by andrsuh** [comment]
Check that the docs clear for you and written correctly please

File: eureka-service-discovery/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Eureka Discovery service
+## Application provides service discovery mechanism - each service knows the addresses of existing Eureka instances and 
+sends some heartbeats to it periodically. So long as Eureka receives these heartbeats from service it is considered to be
```
Comment (shinehlina) [inline eureka-service-discovery/README.md]: AS long as Eureka receives? I'm not sure in "so long as" https://dictionary.cambridge.org/ru/%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B0%D1%82%D0%B8%D0%BA%D0%B0/%D0%B1%D1%80%D0%B8%D1%82%D0%B0%D0%BD%D1%81%D0%BA%D0%B0%D1%8F-%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B0%D1%82%D0%B8%D0%BA%D0%B0/as-and-as-expressions/as-long-as-and-so-long-as

File: eureka-service-discovery/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Eureka Discovery service
+## Application provides service discovery mechanism - each service knows the addresses of existing Eureka instances and 
+sends some heartbeats to it periodically. So long as Eureka receives these heartbeats from service it is considered to be 
+registered. It's easier to think about Eureka like it was just a map where the key is some service name and the value is 
+the list of working instances of this service. Each service may request for the list of working instances by passing the 
+service name.
+
+## Build and run discovery service
+Eureka server is represented by Spring Boot application and is build in executable jar which in turn is packed in docker
```
Comment (shinehlina) [inline eureka-service-discovery/README.md]: is builT in

File: eureka-service-discovery/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Eureka Discovery service
+## Application provides service discovery mechanism - each service knows the addresses of existing Eureka instances and 
+sends some heartbeats to it periodically. So long as Eureka receives these heartbeats from service it is considered to be 
+registered. It's easier to think about Eureka like it was just a map where the key is some service name and the value is 
+the list of working instances of this service. Each service may request for the list of working instances by passing the 
+service name.
+
+## Build and run discovery service
+Eureka server is represented by Spring Boot application and is build in executable jar which in turn is packed in docker 
+image. Eureka server docker image with `eureka-server` tag might be created during the gradle build. in order to create
```
Comment (shinehlina) [inline eureka-service-discovery/README.md]: In order (Capital)

File: eureka-service-discovery/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+## Application provides service discovery mechanism - each service knows the addresses of existing Eureka instances and 
+sends some heartbeats to it periodically. So long as Eureka receives these heartbeats from service it is considered to be 
+registered. It's easier to think about Eureka like it was just a map where the key is some service name and the value is 
+the list of working instances of this service. Each service may request for the list of working instances by passing the 
+service name.
+
+## Build and run discovery service
+Eureka server is represented by Spring Boot application and is build in executable jar which in turn is packed in docker 
+image. Eureka server docker image with `eureka-server` tag might be created during the gradle build. in order to create 
+Eureka image you should run `gradle build buildImage` command from the eureka project root directory. Eureka can be ran
```
Comment (shinehlina) [inline eureka-service-discovery/README.md]: rUn (This is past participle)

**Review by shinehlina — CHANGES_REQUESTED** [review]

File: eureka-service-discovery/README.md (hunk 1 above)
Comment (andrsuh) [reply eureka-service-discovery/README.md]: that means condition. but ok, there is no difference for me :)

**Pull request merged**